crates/sui-indexer/src/models/checkpoints.rs

```diff
@@ -209,3 +209,23 @@ impl TryFrom<StoredCheckpoint> for RpcCheckpoint {
         })
     }
 }
+
+#[derive(Queryable, Insertable, Selectable, Debug, Clone, Default)]
+#[diesel(table_name = cp_tx)]
+pub struct StoredCpTx {
+    pub checkpoint_sequence_number: i64,
+    pub min_tx_sequence_number: i64,
+    pub max_tx_sequence_number: i64,
+}
+
+impl From<&IndexedCheckpoint> for StoredCpTx {
+    fn from(c: &IndexedCheckpoint) -> Self {
+        Self {
+            checkpoint_sequence_number: c.sequence_number as i64,
+            // in practice, there is at least one transaction in a checkpoint that mutates
+            // the Clock object.
+            min_tx_sequence_number: c.min_tx_sequence_number.unwrap_or(0) as i64,
+            max_tx_sequence_number: c.max_tx_sequence_number.unwrap_or(0) as i64,
```

Comment:
im not sure if we should set a default to 0? Yes in practice we have at least one transaction per checkpoint, but this is not always the case in all of our (e2e) tests. Could we instead set the min_tx_sequence_number to `start = ckpt.network_total_transactions - self.size() as u64;` and the max_tx_sequence_number to `self.size() - 1`?

Reply:
Otherwise, if we somehow have a cp with 0 min and max, we'll scan from an erroneous point in time (for example, checkpoint 30 million without txs will cause graphql to start scanning from 0)

Reply:
if the e2e test is not following the production, we should change the e2e test instead?

Reply:
to avoid the unwrap_or, I think we should make it non-optional, as it's very unlikely that we will not update the clock object, if that somehow indeed happened, we prob have more to worry about then the breaking here.

Reply:
Aye agreed

Reply:
cool cool, I will do a quick follow up commit to get rid the optional tx seq all together.